src/dataplane-protocol/src/smartstream.rs

```diff
@@ -0,0 +1,167 @@
+pub use encoding::{
+    SmartStreamRuntimeError, SmartStreamRuntimeErrorBuilder, SmartStreamType, SmartStreamOutput,
+};
+
+pub type Result<T> = std::result::Result<T, SmartStreamRuntimeErrorBuilder>;
+
+mod encoding {
+    use std::fmt;
+    use crate::Offset;
+    use crate::record::Record;
+    use fluvio_protocol::{Encoder, Decoder};
+
+    /// A type used to capture and serialize errors from within a SmartStream
+    #[derive(thiserror::Error, Debug, Default, Clone, PartialEq, Encoder, Decoder)]
+    pub struct SmartStreamRuntimeError {
+        /// Error hint: meant for users, not for code
+        pub hint: String,
+        /// The offset of the Record that had a runtime error
+        pub offset: i64,
+        /// The type of SmartStream that had a runtime error
+        pub kind: SmartStreamType,
+        /// A preview of the Record key that caused this error
+        pub record_key: String,
```

Comment:
This should be RecordData

Reply:
This field is supposed to be a "preview" of the record data in a user-readable way, it is not intended to preserve the exact contents of the record. Notably, this field gets truncated so that it is only 100 columns wide and at most 20 lines long, so that the error type size does not balloon

Reply:
this is supposed to be only dealing with types not user display issue. let CLI or other output worry about it. just pass records having problem

Reply:
We still want this to be truncated to a reasonable size though, right? I don't want to have 1K+ data in these errors

Reply:
any record key/value should fit into a single transmission.  this is not something error handling need to worry about it

Reply:
Alright, will use RecordData